File: 724-find-pivot-index/724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        left_sum = 0
        right_sum = sum(nums[1:])
        if left_sum == right_sum:
            return 0
        n = len(nums)
        for i in range(1, n):
            left_sum += nums[i-1]
            right_sum -= nums[i]
            if left_sum == right_sum:
                return i
        return -1

File: 724-find-pivot-index/724-find-pivot-index.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        let mut left_sum = 0;

            let mut left_sum = 0;

            // find sum of all numbers
            let sum: i32 = nums.iter().sum();

            for i in 0..nums.len() {
                if left_sum == sum - left_sum - nums[i] {
                    return i as i32;
                }
                left_sum += nums[i];
            }
            -1
        
        
    }
}